app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -248,11 +307,86 @@ export default class AMRHeatmapVis extends React.Component {
         onNodeHoverOut: this.onNodeHoverOut,
         onColumnLabelClick: onSampleLabelClick,
         onRowLabelClick: this.onRowLabelClick,
+        columnMetadata: this.getSelectedMetadataFields(), // gets the selected metadata *fields*
+        onColumnMetadataLabelMove: this.onNodeHoverMove,
+        onColumnMetadataLabelHover: this.onMetadataLabelHover,
+        onColumnMetadataLabelOut: this.onMetadataLabelOut,
+        onAddColumnMetadataClick: this.onMetadataAddButtonClick,
+        marginLeft: 75, // our gene names are very short, so this is to prevent metadata names from disappearing
       }
     );
     this.heatmap.start();
   }
 
+  //*** Render functions ***
+
+  renderMetadataLegend() {
```

Comment:
These two components look nearly identical to the ones in `SamplesHeatmapVis` (both the JS and the CSS). Could you extract them into a shared component that both heatmaps can use?

Reply:
Metadata legend and metadata selector.

Reply:
I'll see what I can do!